torch/amp/autocast_mode.py

```diff
@@ -189,37 +192,10 @@ def __init__(self, device_type : str,
             assert dtype is not None
             return
         self.device = device_type
-        self.custom_backend_name = torch._C._get_privateuse1_backend_name()
-        if self.device == 'cuda':
-            self.fast_dtype = torch.get_autocast_gpu_dtype()
-        elif self.device == 'cpu':
-            self.fast_dtype = torch.get_autocast_cpu_dtype()
-        elif self.device == 'xpu':
-            self.fast_dtype = torch.xpu.get_autocast_xpu_dtype()  # type: ignore[attr-defined]
-        elif self.device == 'ipu':
-            self.fast_dtype = torch.get_autocast_ipu_dtype()  # type: ignore[attr-defined]
-        elif self.device == 'hpu':
-            self.fast_dtype = torch.hpu.get_autocast_hpu_dtype()  # type: ignore[attr-defined]
-        elif self.device == 'xla':
-            self.fast_dtype = torch.get_autocast_xla_dtype()  # type: ignore[attr-defined]
-        elif self.device == self.custom_backend_name:
-            necessary_funcs = ['is_autocast_enabled', 'set_autocast_enabled', 'get_autocast_dtype',
```

Comment:
> This changes what we expect custom backends to have to implement for AMP support right?
> 
> * Do previous custom backend module's implementations just end up calling into something like `at::autocast::set_privateuseone_enabled(enabled)`? One risk if the user defined `torch.foo.set_autocast_dtype` does some kind of side effect, that would break their code since it is no longer called.
> * It also looks like we still use `getattr(torch, self.device).get_amp_supported_dtype()` later in this PR, should we keep the proper error checking for that here?
> 
> This does look like it would clean up things, but I would be curious what @bdhirsh thinks.

yeah, I have add the `func` check and testcast for `get_amp_supported_dtype` defined in custom device backend. And as for the func `at::autocast::set_privateuseone_enabled(enabled)`, it is just defined in c++, and it does not export to Python, so I think it is ok. @soulitzer

Reply:
Going to defer to @bdhirsh on this one